internal/tofu/transform_attach_config_resource.go

```diff
@@ -50,63 +51,23 @@ func (t *AttachResourceConfigTransformer) Transform(g *Graph) error {
 			log.Printf("[TRACE] AttachResourceConfigTransformer: %q (%T) has no configuration available", dag.VertexName(v), v)
 			continue
 		}
-
-		for _, r := range config.Module.ManagedResources {
-			rAddr := r.Addr()
-
-			if rAddr != addr.Resource {
-				// Not the same resource
-				continue
-			}
-
-			log.Printf("[TRACE] AttachResourceConfigTransformer: attaching to %q (%T) config from %s", dag.VertexName(v), v, r.DeclRange)
-			arn.AttachResourceConfig(r)
-
-			// attach the provider_meta info
-			if gnapmc, ok := v.(GraphNodeAttachProviderMetaConfigs); ok {
-				log.Printf("[TRACE] AttachResourceConfigTransformer: attaching provider meta configs to %s", dag.VertexName(v))
-				if config == nil {
-					log.Printf("[TRACE] AttachResourceConfigTransformer: no config set on the transformer for %s", dag.VertexName(v))
-					continue
-				}
-				if config.Module == nil {
-					log.Printf("[TRACE] AttachResourceConfigTransformer: no module in config for %s", dag.VertexName(v))
-					continue
-				}
-				if config.Module.ProviderMetas == nil {
-					log.Printf("[TRACE] AttachResourceConfigTransformer: no provider metas defined for %s", dag.VertexName(v))
-					continue
-				}
-				gnapmc.AttachProviderMetaConfigs(config.Module.ProviderMetas)
-			}
+		var m map[string]*configs.Resource
+		if addr.Resource.Mode == addrs.ManagedResourceMode {
+			m = config.Module.ManagedResources
+		} else if addr.Resource.Mode == addrs.DataResourceMode {
+			m = config.Module.DataResources
 		}
-		for _, r := range config.Module.DataResources {
-			rAddr := r.Addr()
-
-			if rAddr != addr.Resource {
-				// Not the same resource
-				continue
-			}
-
-			log.Printf("[TRACE] AttachResourceConfigTransformer: attaching to %q (%T) config from %#v", dag.VertexName(v), v, r.DeclRange)
-			arn.AttachResourceConfig(r)
-
-			// attach the provider_meta info
-			if gnapmc, ok := v.(GraphNodeAttachProviderMetaConfigs); ok {
-				log.Printf("[TRACE] AttachResourceConfigTransformer: attaching provider meta configs to %s", dag.VertexName(v))
-				if config == nil {
-					log.Printf("[TRACE] AttachResourceConfigTransformer: no config set on the transformer for %s", dag.VertexName(v))
-					continue
-				}
-				if config.Module == nil {
-					log.Printf("[TRACE] AttachResourceConfigTransformer: no module in config for %s", dag.VertexName(v))
-					continue
-				}
-				if config.Module.ProviderMetas == nil {
-					log.Printf("[TRACE] AttachResourceConfigTransformer: no provider metas defined for %s", dag.VertexName(v))
-					continue
+		if m != nil {
```

Comment:
Instead of checking if m != nil, you could add an else to the mode check with a panic if the mode is not set to the two valid values.  It's slightly more defensive in case that enum is ever extended.

Reply:
Done